lib/MongoDB/ChangeStream.pm

```diff
@@ -0,0 +1,201 @@
+#
+#  Copyright 2009-2018 MongoDB, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+use strict;
+use warnings;
+package MongoDB::ChangeStream;
+
+# ABSTRACT: A stream providing update information for collections.
+
+use version;
+our $VERSION = 'v1.999.0';
+
+use Moo;
+use Try::Tiny;
+use MongoDB::Cursor;
+use MongoDB::Op::_Aggregate;
+use MongoDB::Error;
+use Safe::Isa;
+use MongoDB::_Types qw(
+    MongoDBCollection
+    ArrayOfHashRef
+);
+use Types::Standard qw(
+    InstanceOf
+    HashRef
+    Str
+);
+
+use namespace::clean -except => 'meta';
+
+has _cursor => (
+    is => 'rw',
+    isa => InstanceOf['MongoDB::QueryResult'],
+    lazy => 1,
+    builder => '_build_cursor',
+    clearer => '_clear_cursor',
+);
+
+has collection => (
+    is => 'ro',
+    isa => MongoDBCollection,
+    required => 1,
+);
+
+has options => (
```

Comment:
Please rename "aggregation_options" to make it clear these are not the exact same as the options passed to `watch`.